CMakeLists.txt

```diff
@@ -420,6 +420,7 @@ macro(MACRO_CORE SCRIPT DEFINE BUILD_DEPRECATED)
         ${TIC80CORE_DIR}/core/draw.c
         ${TIC80CORE_DIR}/core/io.c
         ${TIC80CORE_DIR}/core/sound.c
+        ${TIC80CORE_DIR}/core/sokol_time_impl.c
```

Comment:
TIC-80 core should be independent from 3rd party libs, pls don't use Sokol here.
We have to find another way to measure time, for example we could add `time()` function to the `tic80` struct and implement it somewhere on the rendering side, where we use SDL and Sokol libs.
```c
    struct
    {
        u64 (*time)();

        void (*trace)(const char* text, u8 color);
        void (*error)(const char* info);
        void (*exit)();
    } callback;
```
https://github.com/nesbox/TIC-80/blob/d23ffe1daebf33780f6b8f7bc6d7c83c2716af0c/include/tic80.h#L60

Reply:
I feel like that's a waste of coding time, to make sure every independent system has its own way to get time. No matter what library you use, at the end of the day, you'll end up calling a function that basically does what Sokol's time library does, so why not just... use *that* one and avoid annoying platform-specific timing code?

If you're unwilling to yield on "don't use Sokol" then I'll just close this PR and start over. I personally don't see the point in writing our own function for something that's already a solved issue, but if "make a callback function in every single system TIC-80 supports for getting time" is the solution you genuinely want, I've done way too much in this PR to turn around and do that.